### PR TITLE
Adding the Oven Device on top of 8349 change.

### PR DESCRIPTION
**Pull request by eahove**
This adds the oven device type. This is based on the specification revision addressed in PR #30645 

https://github.com/project-chip/connectedhomeip/pull/30645/commits

### CODE REVIEW

**Review by woody-apple — APPROVED** [review]

File: src/controller/data_model/controller-clusters.matter [hunk 1]
```diff
@@ -2968,6 +2968,72 @@ provisional client cluster Timer = 71 {
...
+  revision 1;
+
+  enum ErrorStateEnum : enum8 {
+    kNoError = 0;
+    kUnableToStartOrResume = 1;
+    kUnableToCompleteOperation = 2;
+    kCommandInvalidInState = 3;
+  }
+
+  enum OperationalStateEnum : enum8 {
```
Comment (andy31415) [inline src/controller/data_model/controller-clusters.matter:2982]: Would it make sense to define a separate enum for this as the values in the spec are slightly different? Specifically I see: Runnint, Preheating, Preheated, CoolingDown and SelfCleaning.

File: src/app/zap-templates/zcl/data-model/chip/operational-state-cluster.xml [hunk 2]
```diff
@@ -19,6 +19,7 @@ limitations under the License.
 
   <enum name="OperationalStateEnum" type="enum8">
     <cluster code="0x0060"/>
+    <cluster code="0x0048"/>
```
Comment (andy31415) [inline src/app/zap-templates/zcl/data-model/chip/operational-state-cluster.xml:22]: src/app/zap-templates/zcl/data-model/chip/operational-state-rvc-cluster.xml does not re-use this enum and instead defines it's own with extra values from the spec. I am inclined to think we should do the same here? 

Any reason to re-use this generic one?

Comment (abeck-whirlpool) [reply src/app/zap-templates/zcl/data-model/chip/operational-state-cluster.xml:22]: We are not adding new states to the enum per https://github.com/CHIP-Specifications/connectedhomeip-spec/issues/8349. We are utilizing only what is in the base Operational State cluster

File: src/controller/data_model/controller-clusters.matter (hunk 1 above)
Comment (abeck-whirlpool) [reply src/controller/data_model/controller-clusters.matter:2982]: The spec is being changed in flight on this PR. https://github.com/CHIP-Specifications/connectedhomeip-spec/pull/8583/files. We are no longer having our own Op States for Oven but instead are utilizing specific verbiage for Phase List

File: src/app/zap-templates/zcl/data-model/chip/operational-state-cluster.xml (hunk 2 above)
Comment (eahove) [reply src/app/zap-templates/zcl/data-model/chip/operational-state-cluster.xml]: This file is no longer part of this PR. This change has already been merged to the master.

File: src/controller/data_model/controller-clusters.matter (hunk 1 above)
Comment (eahove) [reply src/controller/data_model/controller-clusters.matter]: This file is no longer part of this PR. This change has already been merged to the master.

**Review by andy31415 — APPROVED** [review]

**Pull request merged**